src/wasm.cc

```diff
@@ -546,9 +591,9 @@ getOrCreateThreadLocalWasm(const std::shared_ptr<WasmHandleBase> &base_handle,
     if (wasm_handle) {
       return wasm_handle;
     }
-    // Remove stale entry.
-    local_wasms.erase(vm_key);
+    local_wasms.erase(it);
```

Comment:
`vm_key` needs to be removed from `local_wasms_keys`, otherwise those two are going to be out of sync.

Reply:
Those keys will be removed from *_keys queues during the cleanup, when they aren't found in the corresponding cache. Otherwise we'd need a linear key search and remove operations to erase it from the queue.

Reply:
Good point. Thanks!